<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>944</width>
    <height>865</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Test Qt Platform</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>5</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Addrinfo</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Hostname:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="addrinfoEdit"/>
          </item>
          <item>
           <widget class="QPushButton" name="addrinfoButton">
            <property name="text">
             <string>getaddrinfo</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QListWidget" name="addrinfoListWidget"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Http</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Method:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="httpMethodBox">
            <property name="frame">
             <bool>true</bool>
            </property>
            <item>
             <property name="text">
              <string>GET</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>POST</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>HEAD</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>DELETE</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Url:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="httpUrlEdit">
            <property name="text">
             <string>https://httpbin.org/get</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="httpPayloadButton">
            <property name="text">
             <string>payload...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="httpSendButton">
            <property name="text">
             <string>Send</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Request header:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="httpRequestHeadersWidget"/>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Reply headers:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="httpReplyHeadersWidget"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,0,0">
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Content:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="httpProxyButton">
            <property name="text">
             <string>Proxy</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="httpSaveButton">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextBrowser" name="httpContentBroswer"/>
        </item>
        <item>
         <widget class="QLabel" name="httpImageLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>CookieJar</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QTreeWidget" name="cookieWidget">
          <property name="autoExpandDelay">
           <number>-1</number>
          </property>
          <property name="indentation">
           <number>25</number>
          </property>
          <column>
           <property name="text">
            <string>Domain</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Path</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Key</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Value</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Tcp</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Echo Server</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Bound:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="tcpEchoEdit">
             <property name="text">
              <string>0.0.0.0:11451</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="tcpEchoButton">
             <property name="text">
              <string>Start</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Client to test</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Server:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="tcpTestEdit">
               <property name="text">
                <string>127.0.0.1:11451</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="tcpTestButton">
               <property name="text">
                <string>Connect and test</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>DataSize:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="tcpDataSizeBox">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>25565</number>
               </property>
               <property name="value">
                <number>200</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>Count:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="tcpCountBox">
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_12">
               <property name="text">
                <string>Timeout:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="tcoTimeoutBox">
               <property name="minimum">
                <number>100</number>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="value">
                <number>1000</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="tcpRandomBox">
               <property name="text">
                <string>Random Data</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Log:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QListWidget" name="tcpLogWidget"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Console</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QPushButton" name="consoleClearButton">
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="consoleStartButton">
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QListWidget" name="consoleListWidget"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_6">
       <attribute name="title">
        <string>Ws</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>URL:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="wsUrlEdit">
            <property name="text">
             <string>wss://echo.websocket.org/</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="wsOpenButton">
            <property name="text">
             <string>Open</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Received:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QListWidget" name="wsReceivedWidget"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>Send:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QListWidget" name="wsSendWidget"/>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <widget class="QLineEdit" name="wsMessageEdit"/>
              </item>
              <item>
               <widget class="QComboBox" name="wsTypeBox">
                <item>
                 <property name="text">
                  <string>Text</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Binary</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="wsSendButton">
                <property name="text">
                 <string>Send</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>944</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>addrinfoEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>addrinfoButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>575</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>httpUrlEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>httpSendButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>459</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>881</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>consoleClearButton</sender>
   <signal>clicked()</signal>
   <receiver>consoleListWidget</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>456</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wsMessageEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>wsSendButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>626</x>
     <y>814</y>
    </hint>
    <hint type="destinationlabel">
     <x>890</x>
     <y>815</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>